### PR TITLE
updated api endpoint for icfp timetable

### PR DESCRIPTION
**Pull request by J05h-L**
### Context
Story - [AB#191586](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/191586)
Task - [AB#194091](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/194091)

### Change proposed in this pull request
This PR updates the api endpoint for a Financial Plan to contain a property for timetable as required for the icfp timetable page

### Guidance to review 
N/A

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs [hunk 1]
```diff
@@ -9,4 +9,5 @@ public record FinancialPlanRequest
     public decimal? TotalTeacherCosts { get; set; }
     public decimal? TotalNumberOfTeachersFte { get; set; }
     public decimal? EducationSupportStaffCosts { get; set; }
+    public int? Timetable { get; set; }
```
Comment (jrabbott) [inline platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs:12]: Please rename to TimetableLength.

**Review by jrabbott — CHANGES_REQUESTED** [review]

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/DataObjects/FinancialPlanDataObject.cs [hunk 2]
```diff
@@ -18,4 +18,5 @@ public record FinancialPlanDataObject
     [JsonProperty("totalNumberOfTeachersFte")] public decimal? TotalNumberOfTeachersFte { get; set; }
     [JsonProperty("educationSupportStaffCosts")] public decimal? EducationSupportStaffCosts { get; set; }
     [JsonProperty("useFigures")] public bool? UseFigures { get; set; }
+    [JsonProperty("Timetable")] public int? Timetable { get; set; }
```
Comment (jrabbott) [inline platform/src/abstractions/EducationBenchmarking.Platform.Domain/DataObjects/FinancialPlanDataObject.cs:21]: Json property name should be lower camel case.

**Review by jrabbott — CHANGES_REQUESTED** [review]

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs (hunk 1 above)
Comment (J05h-L) [reply platform/src/abstractions/EducationBenchmarking.Platform.Domain/Requests/FinancialPlanRequest.cs]: changed all in latest commit

File: platform/src/abstractions/EducationBenchmarking.Platform.Domain/DataObjects/FinancialPlanDataObject.cs (hunk 2 above)
Comment (J05h-L) [reply platform/src/abstractions/EducationBenchmarking.Platform.Domain/DataObjects/FinancialPlanDataObject.cs]: corrected in latest commit

**Review by jrabbott — APPROVED** [review]

**Review by jrabbott — APPROVED** [review]

**Pull request merged**